File: algorithms/Codility-master/Codility-master/9_MaximumSliceProblem/MaxProfit.js
// 100%
function solution(A) {
    // write your code in JavaScript (Node.js 8.9.4)
    if (A.length < 2) return 0;
    let min_price = Number.MAX_VALUE
    let max_profit = 0;
    for (let i=0; i< A.length; i++){
        min_price = Math.min(min_price, A[i]);
        max_profit = Math.max(max_profit, A[i] - min_price);
    }
    return max_profit;
}

File: algorithms/Codility-master/Codility-master/9_MaximumSliceProblem/MaxProfit.py
# 100%
def solution(A):
    # write your code in Python 3.6
    if len(A) < 2: return 0
    
    min_price = 2*10E4
    max_profit = 0

    for X in A:
        min_price = min(min_price, X)
        max_profit = max(max_profit, X - min_price)

    return max_profit
